### ISSUE TITLE
Increasing Porcupine verbosity level or reporting if currently in speech 

### ISSUE THREAD

**Issue by nicktgn** [Picovoice/Porcupine#36]
#### TL;DR : 
**1) Is there any way to make Porcupine log more useful debugging info to LogCat in Android?**

**2) Does Porcupine internally detects beginning and end of speech in the audio data, similar to how PocketSphinx or Snowboy do this? And if it does, is there any way to get these states?**

#### Issue description:
I'm currently integrating Porcupine into my Android app and the issue that I'm facing is that it doesn't detect anything. 

Porcupine happily reports to logcat that it got initialized correctly with proper parameters:
```
06-19 19:08:21.378 11391-11808/ I/PORCUPINE: [32m [INFO] model file path: /storage/emulated/0/Android/data/com.example.app/files/sync/porcupine/porcupine_params.pv[0m
    [32m [INFO] number of keywords: 1[0m
    [32m [INFO] keyword file path [0]: /storage/emulated/0/Android/data/com.example.app/files/sync/porcupine/models/alexa_android.ppn[0m
    [32m [INFO] sensitivity [0]: 0.500000[0m
```
Sample rate and frame length were correctly set up in according to what porcupine requires using `Porcupine#getFrameLength()` and `Porcupine#getSampleRate()`. Porcupine does not show any errors about sample rate or frame length when I feed it audio data.

Keyword: Alexa (other available keywords) 
OS: Android

PS: here's, to better illustrate idea from the 2nd quesiton, an example of how Snowboy reports if the any speech was detected in current audio frame (taken from [here](https://github.com/Kitt-AI/snowboy/blob/master/examples/Android/SnowboyAlexaDemo/src/ai/kitt/snowboy/audio/RecordingThread.java)):

```
// Snowboy hotword detection.
int result = detector.RunDetection(audioData, audioData.length);

if (result == -2) {
   // post a higher CPU usage:
    // sendMessage(MsgEnum.MSG_VAD_NOSPEECH, null);
} else if (result == -1) {
    sendMessage(MsgEnum.MSG_ERROR, "Unknown Detection Error");
} else if (result == 0) {
     // post a higher CPU usage:
     // sendMessage(MsgEnum.MSG_VAD_SPEECH, null);
} else if (result > 0) {
     sendMessage(MsgEnum.MSG_ACTIVE, null);
     Log.i("Snowboy: ", "Hotword " + Integer.toString(result) + " detected!");
     player.start();
}
```

Thanks.

**Comment by kenarsa** [comment]
Hi Nick,

Below is the answer to your questions first. Then I am going to give you ideas on how to tackle the problem.

1- Is there something specific you have in mind? We do log all error events and also log the arguments into the constructor. If there is something you have in mind do let us know and we see if it makes sense.
2- We do not operate based on a VAD and are using a different set of algorithms. Therefore we cannot provide this information.

Could you please share snippets of the code you are using for integrating Porcupine? Just want to make sure there is not something obvious there.

After reviewing your integration code I think the best way to solve your problem is to record the data fed into porcupine into a file and then make sure that the data is correct. Something similar to what we do in our Python demo here https://github.com/Picovoice/Porcupine/blob/master/demo/python/porcupine_demo.py

**Comment by nicktgn** [comment]
Hi Alireza,

Thanks for your reply and suggestions. I'll try to record and check the data we're feeding to Porcupine first.

**Comment by nicktgn** [comment]
Hi Alireza,

I have finally found a bug in our integration code.

Regarding 1st question, I don't really have any particular request as to what more can Porcupine log. Was just curious if some extra levels of verbosity exist in the Porcupine.

Thanks for your response.
